integ-test/src/test/java/org/opensearch/sql/sql/NestedIT.java

```diff
@@ -0,0 +1,246 @@
+            + "nested(comment.likes) FROM " + TEST_INDEX_NESTED_TYPE;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(6, result.getInt("total"));
+    verifyDataRows(result,
+        rows("e", 1, "a", "ab", 3),
+        rows("f", 2, "b", "aa", 2),
+        rows("g", 1, "c", "aa", 3),
+        rows("h", 4, "c", "ab", 1),
+        rows("i", 5, "a", "ab", 1),
+        rows("zz", 6, "zz", new JSONArray(List.of("aa", "bb")), 10));
+  }
+
+  @Test
+  public void nested_function_mixed_with_non_nested_type_test() {
+    String query =
+        "SELECT nested(message.info), someField FROM " + TEST_INDEX_NESTED_TYPE;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(6, result.getInt("total"));
+    verifyDataRows(result,
+        rows("a", "b"),
+        rows("b", "a"),
+        rows("c", "a"),
+        rows("c", "b"),
+        rows("a", "b"),
+        rows("zz", "a"));
+  }
+
+  @Test
+  public void nested_function_mixed_with_non_nested_types_test() {
+    String query =
+        "SELECT nested(message.info), office, office.west FROM " + TEST_INDEX_MULTI_NESTED_TYPE;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(6, result.getInt("total"));
+    verifyDataRows(result,
+        rows("a",
+            new JSONObject(Map.of("south", 3, "west", "ab")), "ab"),
+        rows("b",
+            new JSONObject(Map.of("south", 5, "west", "ff")), "ff"),
+        rows("c",
+            new JSONObject(Map.of("south", 3, "west", "ll")), "ll"),
+        rows("d", null, null),
+        rows("i", null, null),
+        rows("zz", null, null));
+  }
+
+  @Test
+  public void nested_with_non_nested_type_test() {
+    String query = "SELECT nested(someField) FROM " + TEST_INDEX_NESTED_TYPE;
+
+    Exception exception = assertThrows(RuntimeException.class,
+        () -> executeJdbcRequest(query));
+    assertTrue(exception.getMessage().contains(
+            "{\n" +
+            "  \"error\": {\n" +
+            "    \"reason\": \"Invalid SQL query\",\n" +
+            "    \"details\": \"Illegal nested field name: someField\",\n" +
+            "    \"type\": \"IllegalArgumentException\"\n" +
+            "  },\n" +
+            "  \"status\": 400\n" +
+            "}"
+    ));
+  }
+
+  @Test
+  public void nested_missing_path() {
+    String query = "SELECT nested(message.invalid) FROM " + TEST_INDEX_MULTI_NESTED_TYPE;
+
+    Exception exception = assertThrows(RuntimeException.class,
+        () -> executeJdbcRequest(query));
+    assertTrue(exception.getMessage().contains("" +
+        "{\n" +
+        "  \"error\": {\n" +
+        "    \"reason\": \"Invalid SQL query\",\n" +
+        "    \"details\": \"can't resolve Symbol(namespace=FIELD_NAME, name=message.invalid) in type env\",\n" +
+        "    \"type\": \"SemanticCheckException\"\n" +
+        "  },\n" +
+        "  \"status\": 400\n" +
+        "}"
+    ));
+  }
+
```

Comment:
Add tests with `nested` and `WHERE`, especially with relevance search. Something like:
```sql
SELECT nested(...) FROM ... WHERE match_phrase(...) AND a > 5;
```